src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -390,6 +390,121 @@ BOOST_AUTO_TEST_CASE(SimpleLinearLayerTest)
   BOOST_REQUIRE_EQUAL(arma::accu(delta), 0);
 }
 
+/**
+ * Linear module weight initialization test.
+ */
+BOOST_AUTO_TEST_CASE(LinearLayerWeightInitializationTest)
+{
+  size_t inSize = 10, outSize = 4;
+  Linear<>* linear = new Linear<>(inSize, outSize);
+  linear->Reset();
+  RandomInitialization().Initialize(linear->Weight(), outSize, inSize);
+  linear->Bias().ones();
```

Comment:
Is there any reason you didn't initialize the bias randomly? I am not sure it's really needed, just in case.

Reply:
I meant the following:
```c++
RandomInitialization().Initialize(linear->Bias(), outSize, 1);
```

Reply:
The purpose of this PR is to achieve a way to 
* initialize `weight` and `bias` with different initialization rules. 
* initialize the `weight`/`bias` of a particular layer with different initialization rule than that of the network's.

If we had to initialize `weight` and `bias` with the same initialization rule, we could have done that using
```cpp
RandomInitialization().Initialize(linear->Parameters(), outSize * (inSize + 1), 1)
```
So, it's just to check we are able to use different initialization rules for `weight` and `bias`.